integration/benchmark/cmd/ci/main.go

```diff
@@ -163,57 +170,56 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msgf("unable to init loader")
 	}
+
+	// run load
 	lg.Start()
 
-	testStartTime := time.Now()
+	// prepare data slices
+	dataSlices := make([]dataSlice, 0)
+	go func() {
+		leadDuration, _ := time.ParseDuration(*leadTime)
+		sliceDuration, _ := time.ParseDuration(*sliceSize)
+
+		// wait through lead duration.
+		time.Sleep(leadDuration)
+
+		// grab time into start time
+		startTime := time.Now()
+		// grab total executed transactions into start transactions
+		startExecutedTransactions := lg.GetTxExecuted()
+
+		for !lg.Stopped {
+			time.Sleep(sliceDuration)
+
+			endTime := time.Now()
+			endExecutedTransaction := lg.GetTxExecuted()
+
+			// calculate this slice
+			inputTps := lg.AvgTpsBetween(startTime, endTime)
+			outputTps := float64(endExecutedTransaction-startExecutedTransactions) / sliceDuration.Seconds()
+			slice := dataSlice{
+				GitSha:       gitSha,
+				GoVersion:    goVersion,
+				OsVersion:    osVersion,
+				StartTime:    startTime,
+				EndTime:      endTime,
+				InputTps:     inputTps,
+				OutputTps:    outputTps,
+				BeforeExTxs:  startExecutedTransactions,
+				AfterExTxs:   endExecutedTransaction,
+				RunStartTime: runStartTime}
+
+			startExecutedTransactions = endExecutedTransaction
+			startTime = endTime
+
+			dataSlices = append(dataSlices, slice)
+		}
+	}()
+
 	time.Sleep(loadCase.duration)
```

Comment:
Ideally you do not want to sleep here, but instead select on two channels: 1) test timeout 2) some indication from the loader that it has finished.